src/s3path.jl

```diff
@@ -107,25 +111,35 @@ function Base.tryparse(::Type{<:S3Path}, str::AbstractString; config::Union{Noth
     path = ()
     isdirectory = true
 
-    tokenized = split(str, "/")
-    bucket = strip(tokenized[3], '/')
-    drive = "s3://$bucket"
+    uri = URI(str)
+    params = queryparams(uri)
+    version = get(params, "versionId", nothing)
+    bucket = uri.host
 
-    if length(tokenized) > 3
+    drive = "s3://$bucket"
+    tokenized = isempty(uri.path) ? nothing : split(lstrip(uri.path, '/'), '/')
+    if tokenized !== nothing
         root = "/"
         # If the last tokenized element is an empty string then we've parsed a directory
         isdirectory = isempty(last(tokenized))
-        path = Tuple(filter!(!isempty, tokenized[4:end]))
+        path = Tuple(filter!(!isempty, tokenized))
     end
```

Comment:
I'm not allowed to make this suggestion but everything from `root = ""` can be replaced with:
```julia
uri = URI(str)

drive = "s3://$(uri.host)"
root = isempty(uri.path) ? "" : "/"
isdirectory = isempty(uri.path) || endswith(uri.path, '/')
path = Tuple(split(uri.path, '/'; keepempty=false))
version = get(queryparams(uri), "versionId", nothing)